src/QMCWaveFunctions/OptimizableObject.h

```diff
@@ -72,6 +75,23 @@ class OptimizableObject
   virtual void reportStatus(std::ostream& os) {}
 
   void setOptimization(bool state) { is_optimized_ = state; }
+
+  /** Open HDF file for saving variational parameters */
+  static void openHDFToSave(const std::filesystem::path& filename, hdf_archive& hout);
```

Comment:
If I open file directly and bypass openHDFToSave, VPs can still be read/written. I don't feel OptimizableObject is responsible of versioning VP schemes. If that is really needed, it can be in a separate class with proper encapsulation. In the initial design, prefer to do minimal.